app/models/concerns/orchestration/compute.rb

```diff
@@ -84,6 +84,12 @@ def setUserData
     @host      = self
     # For some reason this renders as 'built' in spoof view but 'provision' when
     # actually used. For now, use foreman_url('built') in the template
+    if template.nil?
+      failure _("#{image.name} needs user data, but <a target='_blank' href='#{url_for(edit_operatingsystem_path(operatingsystem))}'>#{operatingsystem.title}</a> is not associated to any provisioning template of the kind user_data.
```

Comment:
probably `_(%{image} ... %{link} ...) % {:image => ..., :link => ...}` would be better